### PR TITLE
serviceconnector: Add children to validate activity log

### PR DESCRIPTION
**Pull request by motm32**
Here is how the validation success state looks: 
image

Here is how the error state looks: 
image

### CODE REVIEW

File: serviceconnector/src/validateLinker/ValidateLinkerStep.ts [hunk 1]
```diff
@@ -3,20 +3,34 @@
...
+
         for (const detail of nonNullValue(response.validationDetail)) {
-            if (detail.result === KnownValidationResultStatus.Failure) {
+            if (detail.result === "failure") {
+                context.activityChildren.push(new GenericTreeItem(undefined, {
+                    contextValue: `validateResult-${detail.name}`,
+                    label: nonNullValue(detail.name),
+                    iconPath: new ThemeIcon('error', new ThemeColor('testing.iconFailed'))
+                }));
                 throw new Error(detail.description);
```
Comment (alexweininger) [inline serviceconnector/src/validateLinker/ValidateLinkerStep.ts:27]: Although one validation failed, I think it's best to still show all other failed/passed validations as activity children. Maybe throw the error after all the children are added?

File: serviceconnector/src/deleteLinker/IPickLinkerContext.ts [hunk 2]
```diff
@@ -3,9 +3,14 @@
...
+import { ValidationResultItem } from "@azure/arm-servicelinker";
+import { ExecuteActivityContext } from "@microsoft/vscode-azext-utils";
 import { ICreateLinkerContext } from "../createLinker/ICreateLinkerContext";
 
-export interface IPickLinkerContext extends ICreateLinkerContext {
+export interface IPickLinkerContext extends ICreateLinkerContext, ExecuteActivityContext {
     linkerName?: string;
+
+    //Validation
+    validationResult?: ValidationResultItem[]
```
Comment (alexweininger) [inline serviceconnector/src/deleteLinker/IPickLinkerContext.ts:14]: Where is this set or used?

File: serviceconnector/src/validateLinker/ValidateLinkerStep.ts (hunk 1 above)
Comment (motm32) [reply serviceconnector/src/validateLinker/ValidateLinkerStep.ts:27]: Actually tested this out and for some reason wen there is an error the validation result only contains the failed check. So moving the error after all the children doesn't change anything.

Comment (alexweininger) [reply serviceconnector/src/validateLinker/ValidateLinkerStep.ts:27]: I'm not exactly sure why that happens. But that's fine

**Review by alexweininger — APPROVED** [review]

**Pull request merged**